manager/state/raft/raft.go

```diff
@@ -843,13 +849,26 @@ type applyResult struct {
 	err  error
 }
 
-// processInternalRaftRequest sends a message through consensus
-// and then waits for it to be applies to the server. It will
-// block until the change is performed or there is an error
+// processInternalRaftRequest sends a message to nodes participating
+// in the raft to apply a log entry and then waits for it to be applied
+// on the server. It will block until the update is performed, there is
+// an error or until the raft node finalizes all the proposals on node
+// shutdown.
 func (n *Node) processInternalRaftRequest(ctx context.Context, r *api.InternalRaftRequest, cb func()) (proto.Message, error) {
 	r.ID = n.reqIDGen.Next()
 
 	ch := n.wait.register(r.ID, cb)
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
It seems at the moment of `finalize` call either `stopCh` or `doneCh` will be closed so that we can reuse them.